### ISSUE TITLE
non-ascii runes in input cause a panic  

### ISSUE THREAD

**Issue by doooks** [adrg/strutil#18]
I found that certain non-ascii runes in the input to Compare would panic. 

to reproduce

```
j := metrics.NewJaroWinkler()
j.Compare("ab\u2019c", "ab\u2019c")
```

PR incoming.

**Comment by adrg** [comment]
Hi @doooks,

Thank you for the issue and the PR. I already knew about it. There already is another branch on this repository which fixes the problem which I'm going to merge soon. Just need to add some more test cases.

**Comment by adrg** [comment]
The problem has been fixed in release [v0.3.1](https://github.com/adrg/strutil/releases/tag/v0.3.1).
@doooks Thank you for opening the issue.
